File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # Create a dictionary to count how many times number occured
        dic = {}

        for i in range(len(nums)):
            # Check if number is in the dic
            if nums[i] in dic:
                # If it is, increase the value by one
                dic[nums[i]] += 1
            else:
                # If it is, keep it as 1
                dic[nums[i]] = 1
        
        for num in nums:
            # Check if number in the dic occured at many times 
            if dic[num] > 1:
                # If it is, return true
                return True
        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums: number[], k: number): number[] {
        const mapper = new Map<number, { key: number, count: number}>();
        const result = [];

        for (const num of nums) {
            if (!mapper.has(num)) {
                mapper.set(num, { key: num, count: 1 });
                continue;
            }
            mapper.get(num).count += 1;
        }

        let count = 0;
        for (const value of [...mapper.values()].sort((a, b) => b.count - a.count)) {
            if (count === k) break;
            result.push(value.key)
            count++;
        }
        return result;
    }
}
